onadata/apps/logger/migrations/0064_auto_20210304_0314.py

```diff
@@ -0,0 +1,32 @@
+# Generated by Django 2.2.16 on 2021-03-04 08:14
+
+from django.db import migrations
+
+from onadata.apps.logger.models import XFormVersion, XForm
+from onadata.apps.api.tools import create_xform_version
+
+
+def create_initial_xform_version(apps, schema_editor):
+    """
+    Creates an XFormVersion object for an XForm that has no
+    Version
+    """
+    queryset = XForm.objects.filter(
+        downloadable=True,
+        deleted_at__isnull=True
+    )
+    for xform in queryset.iterator():
+        if xform.version and not XFormVersion.objects.filter(
+                xform=xform, version=xform.version).first():
```

Comment:
This seems kind of expensive, for 40k forms this will be 1 select query + 40k select queries + 40k insert queries, could we have a unique filter so that in the event the record exists it will fail and hence have 1 select query + 40k insert queries?

Reply:
Also, this check is also being done by the `create_xform_version()` function, so we currently are going to have 1 select query + 40k select queries + 40k select queries (done by `create_xform_version()`)  + 40k insert queries

Reply:
I've changed this in the latest commit. I've removed the `XFormVersion.objects....` line. In favor of the mechanism in `create_xform_version()`; Just passes on an integrity error....